UPGRADING.md

```diff
@@ -1,6 +1,15 @@
 Upgrading Grape
 ===============
 
+### Upgrading to >= 0.18.1
+
+#### Aligned and clarified callback behavior of 405 handler and built-in OPTIONS handler
+
+When a request is made to a resource using an undefined method (HTTP 405) or a request is
+made to the built-in `OPTIONS` handler, only the `before` callbacks associated with the
+resource will be run.  Other callbacks (`before_validation`, `after_validation`, `after`)
```

Comment:
That is a good point.  If someone tries to build logging using a route-defined ```after``` callback, this will defeat that.

However I think the preferred way to do logging would be to use a logger that inherits from Grape::Middleware::Base.  Both the ```before``` and ```after``` callbacks of middleware are still called even for the built-in ```OPTIONS``` handler, just as they are when an HTTP 405 is generated because of an invalid method.